### PR TITLE
Move ZAI SAPI to root directory

### PR DESCRIPTION
**Pull request by SammyK**
### Description

ZAI SAPI is used to test ZAI seams exclusively from the `zend_abstract_interface/` directory, but there is need for it outside of ZAI seams. This PR makes ZAI SAPI available to other projects by moving it into the root project directory.

- Link ZAI SAPI into a project with CMake (`ZaiSapi_ROOT` must be set to the ZAI SAPI installation path)

```CMake
find_package(ZaiSapi 0.1.0 REQUIRED)
# ...
target_link_libraries(foo PUBLIC ZaiSapi::ZaiSapi)
```
- The recommended install path is `​​/opt/zai_sapi/<php-build-name>`
- ZAI SAPI is built as a static library: `libZaiSapi.a`

There will be a follow up PR to rename ZAI SAPI to 🍵 Tea Sapi (Testing and Embedded Applications SAPI).

## Future scope

I would also like to improve how we're currently managing `PHP_LIB` in CMake so that we can instead do something like:

```CMake
find_package(php 8.0 REQUIRED)
# ...
target_link_libraries(foo PUBLIC php::php)
```

### Reviewer checklist
- [ ] Appropriate labels assigned.
- [ ] Milestone is set.
- [ ] Changelog has been added to the release document. For community contributors the reviewer is in charge of this task.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1204,6 +1204,7 @@ jobs:
             cd /tmp && curl -OL https://github.com/Kitware/CMake/releases/download/v<< parameters.cmake_version >>/cmake-<< parameters.cmake_version >>-Linux-x86_64.tar.gz
             sudo mkdir -p /opt/cmake/<< parameters.cmake_version >>
             cd /opt/cmake/<< parameters.cmake_version >> && sudo tar -xf /tmp/cmake-<< parameters.cmake_version >>-Linux-x86_64.tar.gz --strip 1
+            PATH="/opt/cmake/<< parameters.cmake_version >>/bin:$PATH"
```
[the remark this answers is not in the document]
Comment (SammyK) [reply .circleci/config.yml:1207]: Nice catch. I didn't notice that the CMake path was being set in the Catch2 job. I'll remove it from that one and keep this one.

File: zend_abstract_interface/CMakeLists.txt [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
 option(BUILD_ZAI_TESTING "Enable tests" OFF)
-if (${BUILD_ZAI_TESTING})
+if(${BUILD_ZAI_TESTING})
   # Tests uses the C++ testing framework Catch2
   enable_language(CXX)
 
-  # The Catch2::Catch2 target has been available since 2.1.2
-  # We are unsure of the true minimum, but have tested 2.4
+  # The Catch2::Catch2 target has been available since 2.1.2 We are unsure of
+  # the true minimum, but have tested 2.4
```
[the remark this answers is not in the document]
Comment (SammyK) [reply zend_abstract_interface/CMakeLists.txt:15]: Yes, that's the work of `cmake-format`.

**Review by morrisonlevi — APPROVED** [review]
At least as far as I can tell from just reading it, it looks good to me. The true test will come when we try to write new tests against it that aren't ZAI related at all.

File: .circleci/config.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (SammyK) [reply .circleci/config.yml:1207]: Addressed in #1443.

**Pull request merged**